include/flatbuffers/util.h

```diff
@@ -45,13 +50,63 @@
 
 namespace flatbuffers {
 
+#ifdef FLATBUFFERS_PREFER_PRINTF
+template<typename T> size_t IntToDigitCount(T t) {
+  size_t digit_count = 0;
+
+  // Count the sign for negative numbers
+  if (t < 0) {
+    digit_count++;
+  }
+
+  // Count a single 0 left of the dot for fractional numbers
+  if (-1 < t && t < 1) {
+    digit_count++;
+  }
+
+  // Count digits until fractional part
+  T eps = std::numeric_limits<float>::epsilon();
+  while (t <= (-1 + eps) || (1 - eps) <= t) {
+    t /= 10;
+    digit_count++;
+  }
+
+  return digit_count;
+}
+
+template<typename T> size_t NumToStringWidth(T t, int precision = 0) {
+  size_t string_width = IntToDigitCount(t);
+  if (precision) {
+    string_width += (precision + 1); // Count the dot for floating point numbers
+  }
+
+  return string_width;
+}
+
+template<typename T> std::string NumToStringImplWrapper(T t, const char* fmt,
+                                                        int precision = 0) {
+  size_t string_width = NumToStringWidth(t, precision);
+  std::string s(string_width+1, 0x00); // Allocate an extra null byte
```

Comment:
`std::string` already comes with null-termination built-in (right after `size()`, so I don't think the +1 is needed.

Reply:
Well that sounds great, I was vaguely aware of that but I thought it was an implementation detail. Removed (and removed the `resize` -1 later) in e04a8463